app/main/tests/tests_views.py

```diff
@@ -0,0 +1,48 @@
+from http import HTTPStatus
+
+from django.contrib.auth.models import User
+from django.test import TestCase
+from django.urls import reverse
+from django.utils.http import urlencode
+
+from app.main.factories import StaffUserFactory
+
+
+class TestIndexView(TestCase):
+    @classmethod
+    def setUpTestData(cls):
+        cls.admin_user = StaffUserFactory()
+
+    def test_get_index_not_authorised(self):
+        response = self.client.get(reverse("main:index"))
+        login_url = reverse("admin:login") + "?" + urlencode({"next": "/"})
+        self.assertRedirects(response, login_url, HTTPStatus.FOUND)
```

Comment:
302 is always the default, I personally wouldn't bother being explicit about it. It feels like it would be very rare for this to be 301.

Reply:
Makes sense!